internal/job/executor.go

```diff
@@ -850,18 +850,40 @@ func (e *Executor) VendoredPluginPhase(ctx context.Context) error {
 	return e.executePluginHook(ctx, "environment", vendoredCheckouts)
 }
 
+var singleHooks = map[string]bool{
```

Comment:
I think you need to give this a better name, or explain how it is used with a comment. Using the name of the experiment, `strict-single-hooks`, in some way would make this a bit more clear.